External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp

```diff
@@ -981,21 +981,37 @@ void JITCore::CreateCustomDispatch(FEXCore::Core::InternalThreadState *Thread) {
     }
   };
 
+  aarch64::Label FullLookup{};
+
   // We want to ensure that we are 16 byte aligned at the top of this loop
   Align16B();
 
   bind(&LoopTop);
   AbsoluteLoopTopAddress = GetLabelAddress<uint64_t>(&LoopTop);
 
-  // This is the block cache lookup routine
-  // It matches what is going on it BlockCache.h::FindBlock
-  ldr(x0, &l_PagePtr);
-
   // Load in our RIP
   // Don't modify x2 since it contains our RIP once the block doesn't exist
   ldr(x2, MemOperand(STATE, offsetof(FEXCore::Core::ThreadState, State.rip)));
   auto RipReg = x2;
 
+  // L1 Cache
+  LoadConstant(x0, Thread->BlockCache->GetL1Pointer());
+
+  and_(x3, RipReg, 1 * 1024 * 1024 - 1);
```

Comment:
Don't use hard coded L1C size.
Pull a constant from the BlockCache class instead

Reply:
Done